app.py

```diff
@@ -1171,12 +1195,14 @@ def export_gpx():
                 for coord in gps_data.get("coordinates", []):
                     if isinstance(coord, list) and len(coord) >= 2:
                         lon, lat = coord[0], coord[1]
-                        gpx_segment.points.append(gpxpy.gpx.GPXTrackPoint(lat, lon))
+                        gpx_segment.points.append(
+                            gpxpy.gpx.GPXTrackPoint(lat, lon))
             elif gps_data.get("type") == "Point":
                 coord = gps_data.get("coordinates", [])
                 if isinstance(coord, list) and len(coord) >= 2:
                     lon, lat = coord[0], coord[1]
-                    gpx_segment.points.append(gpxpy.gpx.GPXTrackPoint(lat, lon))
+                    gpx_segment.points.append(
+                        gpxpy.gpx.GPXTrackPoint(lat, lon))
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider consolidating duplicate GPX point logic</b></div>
<div id="fix">

Consider consolidating the duplicate GPX point creation logic in `export_gpx()`. The same code for creating track points appears in both the LineString and Point cases.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<blockquote>Check the AI-generated fix before applying</blockquote>
<div id="code">


```
 @@ -1174,6 +1174,10 @@
 +    def add_gpx_point(segment, lat, lon):
 +        segment.points.append(gpxpy.gpx.GPXTrackPoint(lat, lon))
 +
      if gps_data.get("type") == "LineString":
          for coord in gps_data.get("coordinates", []):
              if isinstance(coord, list) and len(coord) >= 2:
                  lon, lat = coord[0], coord[1]
 -                gpx_segment.points.append(gpxpy.gpx.GPXTrackPoint(lat, lon))
 +                add_gpx_point(gpx_segment, lat, lon)
      elif gps_data.get("type") == "Point":
          coord = gps_data.get("coordinates", [])
          if isinstance(coord, list) and len(coord) >= 2:
              lon, lat = coord[0], coord[1]
 -            gpx_segment.points.append(gpxpy.gpx.GPXTrackPoint(lat, lon))
 +            add_gpx_point(gpx_segment, lat, lon)
```

</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/realronaldrump/everystreet-new/pull/95#issuecomment-2543418263>#d8630f</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged